### ISSUE TITLE
Suggestion: user should choose the positioning of the tooltip

### ISSUE THREAD

**Issue by mattiaerli97** [bumbeishvili/d3-v6-tip#7]
By default, the tooltip is positioned as 'absolute' as reported in the following snippets (not in the same position in the original code):
```
tipNode
      .classed('d3-tip-lib-node', true)
      .style('position', 'absolute')
      .style('top', '0')
      .style('opacity', '0')
      .style('pointer-events', 'none')
      .style('box-sizing', 'border-box')


nodel.html(content)
            .style('position', 'absolute')
            .style('opacity', 1)
            .style('pointer-events', 'all')
```

This positioning, however can lead to different problems, as, for example, if the chart is contained in a 'relative' container, the position of the tooltip isn't correct. In the example, a 'fixed' positioning should be considered.

What about adding a custom option to set it absolute? Thank you in advance and I hope to do something for you that can be useful

**Comment by bumbeishvili** [comment]
@mattiaerli97 can you provide the following examples?

1. Example where the problem is reproduced 
2. Example where the problem is fixed based on your code additions

I myself faced a problem when inner SVG elements were scaled and rotated, the tip was showing up at the wrong (undesired) location, but haven't had other positioning problems with it

**Comment by mattiaerli97** [comment]
In the following screen I had a 'relative' positioning in the first element you can see in the console, so the tooltip, when we hover on Sudafrica, is not correctly positioned

![image](https://user-images.githubusercontent.com/45450225/118794311-bfa99080-b899-11eb-993c-45ae66dd3a06.png)

When I add a 'fixed' positioning to the tip (in this case I used an !important selector but only because the 'absolute' style is inline and it's the only way I can override it), all the visualization is correct

![image](https://user-images.githubusercontent.com/45450225/118794748-2b8bf900-b89a-11eb-8634-9e4e6d1c0104.png)

**Comment by bumbeishvili** [comment]
@mattiaerli97 can you provide a live (jsfiddle) example so I can take a look at the full code?

**Comment by mattiaerli97** [comment]
I don't know how to reproduce this in a jsfiddle, because I used all my company's packages to make the screenshots and they aren't published to npm, so I can't include them.

**Comment by bumbeishvili** [comment]
I understand that, but I also can't be sure that it was a d3-tip issue unless I will check the live reproduced example

**Comment by mattiaerli97** [comment]
https://jsfiddle.net/mattiaerli97/qhycz67m/25/

In this jsfiddle I:
1) Included the current version with the new rootElement(), which is 1.0.9
2) I created a div with a relative positioning and included into it the svg
3) in the same div element (so at the same svg level) I attached the tooltip with the rootElement() method

With the position 'absolute' this can't working well, in fact in the fiddle, the tooltip isn't working correctly

**Comment by bumbeishvili** [comment]
It looks like rootElement causes that behavior, if you disable rootElement it works correctly after that

**Comment by mattiaerli97** [comment]
Yes that's correct, but a user might want to attach a tooltip where he has a relative positioning, causing the bug.
In my opinion it's necessary to grant the user to set the tip positioning to other values other that 'absolute'.
If this isn't relevant, I will close the 'issue'

**Comment by bumbeishvili** [comment]
> Yes that's correct, but a user might want to attach a tooltip where he has a relative positioning, causing the bug.
> In my opinion it's necessary to grant the user to set the tip positioning to other values other that 'absolute'.
> If this isn't relevant, I will close the 'issue'

I agree, I just want to make sure that it was not working well for original tip
